json/src/machine.nr

```diff
@@ -233,7 +233,7 @@ pub fn rewrite_stack(input: RewriteStackInput) -> RewriteStackOutput {
     let is_object_value = current_value == [Field::from(1), Field::from(1)];
     let is_array = current_value[0] == Field::from(2);
 
-    let not_to_hash = (parsing_string & next_parsing_string) | next_parsing_primitive;
+    let not_to_hash = !((parsing_string & next_parsing_string) | next_parsing_primitive);
```

Comment:
The original implementation uses `IsZero`, so I appended `!`. In terms of the logic, `state_hash` is accumulated as the parser processes each character, so `parsing_string` requires additional hash update. 
https://github.com/pluto/web-prover-circuits/blob/c0b31bf925fb8a0bec939e8b8167d1a8f8ad2703/circuits/json/hash_machine.circom#L377

This is a part of log from the circom test using `{"outer":{"inner":"value"}}` from our `test_extract_nested_object` test.
```
data_idx 2
next_monomial 1
option_hash 111
to_change_zeroth 1
to_change_first 0
i 0
next_tree_hash[i][0] 111
next_tree_hash[i][1] 0
i 1
next_tree_hash[i][0] 0
next_tree_hash[i][1] 0
```